src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/SafeHandles.Linux.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+
+namespace System.DirectoryServices.Protocols
+{
+    internal sealed class ConnectionHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        internal bool _needDispose = false;
+
+        internal ConnectionHandle()
+            :base(true)
+        {
+            OpenLDAP.ldap_initialize(out IntPtr ldap, null);
```

Comment:
Why not

```suggestion
            OpenLDAP.ldap_initialize(out handle, null);
```

Reply:
Or, declare ldap_initialize as outing a SafeConnectionHandle instead of an IntPtr...

Reply:
Is it safe to just set handle directly? Does SetHandle do any other validation? If not then I'm fine with doing out handle, and removing the SetHandle

Reply:
just checked here and it looks like we don't do any validation https://source.dot.net/#System.Private.CoreLib/SafeHandle.cs,78
so I'm fine with just doing out handle then